dev-libs/boost/boost-1.88.0.ebuild

```
@@ -0,0 +1,480 @@
+
+	# Some test suites have no main because normally boost.test can
+	# automatically initialize & run them, but this only seems to be
+	# supported for statically linked builds/tests.
+	# Therefore we use an explicit list of tests which need patching
+	# with an additional main().
+	# Determining this dynamically is not really possible.
+	local libs_needpatch=(
+		"accumulators"
+	)
+
+	einfo "Patching:" ${libs_needpatch[@]}
+
+	local lib
+	for lib in "${libs_needpatch[@]}"; do
+		# move into library test dir
+		pushd "${BUILD_DIR}/libs/${lib}/test" >/dev/null || die
+			# find all test cases and patch them
+			local testcases testcase
+			readarray -t testcases < <(find . -name "*.cpp")
+			for testcase in "${testcases[@]}"; do
+				# add main() to bootstrap old-style test suite
+				cat "${FILESDIR}/unit-test-main.cpp" >> ${testcase} || die
+			done
+		popd >/dev/null
+	done
+
+	##
+	## Test exclusions
+	##
+
+	# The following libraries do not compile or fail their tests:
+	local libs_excluded=(
+		# fails to use std::reverse_copy
+		"algorithm"
+		# test output comparison failure
+		"config"
+		# "C++03 support was deprecated in Boost.Chrono 1.82" ??
+		"contract"
+		# undefined reference to `boost::math::concepts::real_concept boost::math::bernoulli_b2n<boost::math::concepts::real_concept>(int)
+		"math"
+		# assignment of read-only member 'gauss::laguerre::detail::laguerre_l_object<T>::order'
+		"multiprecision"
+		# uint8_t is not a member of std
+		"mysql"
+		# PyObject* boost::parameter::python::aux::unspecified_type():
+		#   /usr/include/python3.13/object.h:339:30: error: lvalue required as left operand of assignment
+		"parameter_python"
+		# scope/lambda_tests22.cpp(27): test 'x == 1' failed in function 'int main()'
+		"phoenix"
+		# Unable to find file or target named (yes, really)
+		"predef"
+		# AttributeError: property '<unnamed Boost.Python function>' of 'X' object has no setter
+		"python"
+		# vec_access.hpp:95:223: error: static assertion failed: Boost QVM static assertion failure
+		"qvm"
+		# regex_timer.cpp:19: ../../../boost/timer.hpp:21:3: error: #error This header is
+		#   deprecated and will be removed. (You can define BOOST_TIMER_ENABLE_DEPRECATED to suppress
+		#   this error.)
+		"regex"
+		# in function `boost::archive::tmpnam(char*)': test_array.cpp:(.text+0x108):
+		#   undefined reference to `boost::filesystem::detail::unique_path(...)'
+		"serialization"
+		# TuTestMain.cpp(22) fatal error: in "test_main_caller( argc_ argv )":
+		#   std::runtime_error: Event was not consumed!
+		"statechart"
+		# erase_tests.cpp:(.text+0x44cce): undefined reference to
+		#   tbb::detail::r1::execution_slot(tbb::detail::d1::execution_data const*)
+		"unordered"
+		# t_5_007.cpp(22): error: could not find include file: boost/version.hpp
+		"wave"
+	)
+
+	if ! use mpi; then
+		# graph_parallel tries to use MPI even with use=-mpi
+		local no_mpi=( "mpi" "graph_parallel" )
+		einfo "Disabling tests due to USE=-mpi:" ${no_mpi[@]}
```

```suggestion
		einfo "Disabling tests due to USE=-mpi: ${no_mpi[@]}"
```
... and elsewhere, pls

---

done